org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/IDebugSession.java

```diff
@@ -0,0 +1,38 @@
+package org.eclipse.jdt.ls.debug;
+
+import java.util.function.Consumer;
+
+import org.eclipse.debug.core.DebugEvent;
+import com.sun.jdi.ThreadReference;
+import com.sun.jdi.event.Event;
+import com.sun.jdi.event.StepEvent;
+
+public interface IDebugSession {
+    void suspend();
+
+    void resume();
+
+    static IDebugSession launch(/* TODO: arguments? */) {
+        throw new UnsupportedOperationException();
+    }
+
+    static IDebugSession attach(/* TODO: arguments? */) {
+        throw new UnsupportedOperationException();
+    }
+
+    void detach();
+
+    void terminate();
+
+    // breakpoints
+    void addBreakpoint(String className, int lineNumber, Consumer<IBreakpoint> validatedHandler, Consumer<DebugEvent> breakEventHandler);
```

Comment:
I think we can have a BreakPointUtils to convert (String className, int lineNumber,...) into an IBreakpoint?